test/0.4.24/lidooracle.test.js

```diff
@@ -423,4 +443,45 @@ contract('LidoOracle', ([appManager, voting, user1, user2, user3, user4, nobody]
       })
     })
   })
+
+  describe('Quorum tests', () => {
+    it('all oracles reports random data', async () => {
+      const ETH = (value) => web3.utils.toWei(value + '', 'ether')
+      const oracles = accounts
+      let receipt
+      let i
+
+      await app.setTime(1606824000)
+
+      // Add 10 oracle members
+      for (i = 0; i < 10; i++) {
+        await app.addOracleMember(oracles[i], { from: voting })
+      }
+
+      // Set quorum value to 3
+      await app.setQuorum(3, { from: voting })
+
+      // All oracles are broken and provide random data. The first 8 oracles report
+      // different data.
+      for (i = 0; i < 8; i++) {
+        await app.reportBeacon(0, i, i, { from: oracles[i] })
+      }
+
+      await assertReportableEpochs(0, 0)
+
+      // But the 8th and the 9th oracles randomly report same values. We should not use
+      // that random matched value as quorum!
+      await app.reportBeacon(0, 7, 7, { from: oracles[8] })
+      await assertReportableEpochs(0, 0)
+
+      // The 10th oracle reports same value as the 8th and the 9th.
+      // But we should accept value that was received certain number of times,
+      // because obviously probabilty of this case much less than probability
+      // of 2 random matches. Such quorum mechanics more safe than calculating
+      // mode when certain number of reports recieved.
```

Comment:
Too overcomplicated. The explanation should be simpler:

Previous implementation:
if N members submitted some random garbage and the array is unimodal (contains duplicate reports), it gets pushed (once N >= quorum).

New implementation:
Report get pushed to lido only if N+ members submitted strictly same numbers.

Test checks that previous behaviour got fixed (unimodal data doesn'r get submitted until the quantity of equal reports equals quorum)